src/plugin.js

```diff
@@ -101,44 +161,77 @@ class PluginLoader {
+ * 
+ * Optionally, child classes which extend this can simply define the `exec` function as their own property.
+ * This is the preferred implementation when a custom plugin (subclass) has its own state or additional methods
+ * that are required.
+ * 
+ * @param {function} exec - A function that:
+ *   - Takes in an Express Request object as the first parameter (`req`).
+ *   - Takes in an Action object as the second parameter (`action`).
+ *   - Returns a Promise that resolves to an Action.
+ */
+  constructor(exec) {
+    super();
+    this.isGitProxyPushActionPlugin = true;
+    this.exec = exec;
+  }
+}
 
 /**
- * A plugin which executes a function when receiving a proxy request.
+ * A plugin which executes a function when receiving a git fetch request.
  */
-class ActionPlugin extends ProxyPlugin {
+class PullActionPlugin extends ProxyPlugin {
   /**
-   * Custom function executed as part of the action chain. The function
-   * must take in two parameters, an {@link https://expressjs.com/en/4x/api.html#req Express Request}
-   * and the current Action executed in the chain.
-   * @param {Promise<Action>} exec A Promise that returns an Action &
-   *                               executes when a push is proxied.
+   * Wrapper class which contains at least one function executed as part of the action chain for git pull operations.
+   * The function must be called `exec` and take in two parameters: an Express Request (req) and the current Action
+   * executed in the chain (action). This function should return a Promise that resolves to an Action.
+   * 
+   * Optionally, child classes which extend this can simply define the `exec` function as their own property.
+   * This is the preferred implementation when a custom plugin (subclass) has its own state or additional methods
+   * that are required.
+   * 
+   * @param {function} exec - A function that:
+   *   - Takes in an Express Request object as the first parameter (`req`).
+   *   - Takes in an Action object as the second parameter (`action`).
+   *   - Returns a Promise that resolves to an Action.
    */
   constructor(exec) {
     super();
+    this.isGitProxyPullActionPlugin = true;
     this.exec = exec;
   }
 }
 
-const createLoader = async () => {
-  // Auto-register plugins that are part of git-proxy core
-  let names = [];
-  let files = [];
-  if (process.env.GITPROXY_PLUGIN_PACKAGES !== undefined) {
-    names = process.env.GITPROXY_PLUGIN_PACKAGES.split(',');
-  }
-  if (process.env.GITPROXY_PLUGIN_FILES !== undefined) {
-    files = process.env.GITPROXY_PLUGIN_FILES.split(',');
-  }
-  const loader = new PluginLoader(names, files);
-  if (names.length + files.length > 0) {
-    loader.load();
-  }
+/**
+ * 
+ * @param {Array<string>} targets A list of loadable targets for plugin modules. 
+ * @return {PluginLoader}
+ */
+const createLoader = async (targets) => {
+  const loadTargets = targets;
```

Comment:
minor nitpick but do we need to reassign the targets to loadTargets here? Can we not directly pass it on PluginLoader?

Reply:
Nope - this is left over from a previous implementation so it can be removed. Agree that it's extraneous.